pkg/bgpv1/gobgp/state.go

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package gobgp
+
+import (
+	"context"
+	"time"
+
+	gobgp "github.com/osrg/gobgp/v3/api"
+
+	"github.com/cilium/cilium/api/v1/models"
+)
+
+// GetPeerState invokes goBGP ListPeer API to get current peering state.
+func (sc *ServerWithConfig) GetPeerState(ctx context.Context) ([]*models.BgpPeer, error) {
+	var data []*models.BgpPeer
+	fn := func(peer *gobgp.Peer) {
+		if peer == nil {
+			return
+		}
+
+		peerState := &models.BgpPeer{}
+
+		if peer.Conf != nil {
+			peerState.LocalAsn = int64(peer.Conf.LocalAsn)
+			peerState.PeerAddress = peer.Conf.NeighborAddress
+			peerState.PeerAsn = int64(peer.Conf.PeerAsn)
+		}
+
+		if peer.State != nil {
+			peerState.SessionState = peer.State.SessionState.String()
+			peerState.AdminState = peer.State.AdminState.String()
```

Comment:
We don't have any feature to set the admin state right now, so IMO, we don't have to expose this now.